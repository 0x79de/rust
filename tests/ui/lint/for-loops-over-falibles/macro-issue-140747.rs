#![forbid(for_loops_over_fallibles)]

fn main() {
    macro_rules! x {
        () => {
            None::<i32> //~ ERROR for loop over an `Option`. This is more readably written as an `if let` statement [for_loops_over_fallibles]
        };
    }
    for _ in x! {} {}
}
