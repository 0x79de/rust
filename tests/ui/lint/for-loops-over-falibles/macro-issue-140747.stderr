error: for loop over an `Option`. This is more readably written as an `if let` statement
  --> $DIR/macro-issue-140747.rs:6:13
   |
LL |             None::<i32>
   |             ^^^^^^^^^^^
...
LL |     for _ in x! {} {}
   |              ----- in this macro invocation
   |
note: the lint level is defined here
  --> $DIR/macro-issue-140747.rs:1:11
   |
LL | #![forbid(for_loops_over_fallibles)]
   |           ^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in the macro `x` (in Nightly builds, run with -Z macro-backtrace for more info)
help: to check pattern in a loop use `while let`
   |
LL ~             ) =
LL |         };
LL |     }
LL ~     while let Some(_ in x! {} {}
   |
help: consider using `if let` to clear intent
   |
LL ~             ) =
LL |         };
LL |     }
LL ~     if let Some(_ in x! {} {}
   |

error: aborting due to 1 previous error

